### ISSUE TITLE
Use multiple provider in structured rails app

### ISSUE THREAD

**Issue by Tamiyadd** [shakacode/react_on_rails#194]
Hi all, 
i'm using this awesome gem for a bit and it's very usefull but i didn't expected how to have multiple component add in a page connected to the redux-store.

For example i need to have component A and B in a page so i use 
```
<%= react_component('A', @resource_props, {prerender: false}) %>
```
and
```
<%= react_component('B', @resource_props, {prerender: false}) %>
```
to add them in my page. I need to have both of them connected to redux-store so i made two containers

```
export default class A extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Awidget></Awidget>
      </Provider>
    );
  }
}

export default class B extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Bwidget></Bwidget>
      </Provider>
    );
  }
}
```
and two component exported with the redux connect method. Everything works fine except that i need to create the store in my store.jsx file with
```
export const store = configureStore();
```
and import it in two container to have them both access the store but this prevent me to initialize my store with some props from Rails like in the react_on_rails helloworld example. How can i pass the store to both the Provider and initialize it with some props? I'm doing something wrong?

I hope I have been clear and thanks for the help

**Comment by justin808** [comment]
@alexfedoseev should this be feasible? 

@Tamiyadd Can you setup your import of the store so that when you import it, it lazily loads.

**Comment by justin808** [comment]
@alexfedoseev and I discussed this. I'll be coming up with a solution soon.

**Comment by robwise** [comment]
For those following this issue, keep an eye on https://github.com/shakacode/react_on_rails/pull/215

**Comment by justin808** [comment]
I hope to have this done by the weekend.

**Comment by justin808** [comment]
This has shipped in 3.0.0.rc.1, @Tamiyadd
